### PR TITLE
fix(networkmonitor): break import dependency with wakunode2 app

### PR DESCRIPTION
**Pull request by LNSD**
Importing "app" (or tool) modules within other modules is not a good practice. App modules should be final (roots of the imports tree). This PR breaks the `networkmonitor` tool's import dependency on the `wakunode2` app.

- [x] Copied the `retrieveDynamicBootstrapNodes` from `wakunode2` to the `networkmonitor` module to break the dependency.
- [x] Fixed a couple of minor deprecation warnings (e.g., use `Duration` as `asyncSleep` argument).

### CODE REVIEW

File: tools/networkmonitor/networkmonitor.nim [hunk 1]
```diff
@@ -193,7 +193,33 @@ proc crawlNetwork(node: WakuNode,
     # we dont run ipMajorityLoop
     # we dont run revalidateLoop
 
-    await sleepAsync(crawlInterval)
+    await sleepAsync(crawlInterval.millis)
+
+proc retrieveDynamicBootstrapNodes(dnsDiscovery: bool, dnsDiscoveryUrl: string, dnsDiscoveryNameServers: seq[ValidIpAddress]): Result[seq[RemotePeerInfo], string] =
```
Comment (rymnc) [inline tools/networkmonitor/networkmonitor.nim:198]: maybe a comment here that this is the same as the proc in wakunode2?

**Review by rymnc — APPROVED** [review]
LGTM

File: tools/networkmonitor/networkmonitor.nim [hunk 2]
```diff
@@ -193,7 +193,33 @@ proc crawlNetwork(node: WakuNode,
...
+      nameServers.add(initTAddress(ip, Port(53))) # Assume all servers use port 53
+
+    let dnsResolver = DnsResolver.new(nameServers)
+
+    proc resolver(domain: string): Future[string] {.async, gcsafe.} =
+      trace "resolving", domain=domain
+      let resolved = await dnsResolver.resolveTxt(domain)
+      return resolved[0] # Use only first answer
+
+    var wakuDnsDiscovery = WakuDnsDiscovery.init(dnsDiscoveryUrl, resolver)
```
Comment (Ivansete-status) [inline tools/networkmonitor/networkmonitor.nim:214]: Not bit deal but shall use let instead?
```suggestion
    let wakuDnsDiscovery = WakuDnsDiscovery.init(dnsDiscoveryUrl, resolver)
```

**Review by Ivansete-status — APPROVED** [review]

**Review by alrevuelta — APPROVED** [review]

File: tools/networkmonitor/networkmonitor.nim (hunk 1 above)
Comment (LNSD) [reply tools/networkmonitor/networkmonitor.nim:198]: This method probably will change in `wakunode2` app after the refactoring feature branch is merged into the `master` branch. So... I don't think a comment will add any value 🤷🏼

File: tools/networkmonitor/networkmonitor.nim (hunk 2 above)
Comment (LNSD) [reply tools/networkmonitor/networkmonitor.nim]: Actually, this broke the build 🤣 Reverting it!

**Comment by LNSD** [comment]
Merging this PR since [the CI checks failure](https://github.com/waku-org/nwaku/actions/runs/4745380376/jobs/8427616948?pr=1692#step:5:4306) is unrelated to this PR's changes.

cc @alrevuelta

**Pull request merged**